### PR TITLE
pkg/tls: cleanup and add OwnerReferences

### PR DESCRIPTION
**Pull request by estroz**
**Description of the change:**
* English corrections and comment cleanup
* Add OwnerReferences to created Secret and ConfigMap objects

**Motivation for the change:** tackle TODO's and make comments easier to read.

### CODE REVIEW

File: pkg/tls/tls.go [hunk 1]
```diff
@@ -139,14 +150,18 @@ type SDKCertGenerator struct {
 	KubeClient kubernetes.Interface
 }
 
-// GenerateCert returns a secret containing the TLS encryption key and cert,
-// a ConfigMap containing the CA Certificate and a Secret containing the CA key or it
-// returns a error incase something goes wrong.
-func (scg *SDKCertGenerator) GenerateCert(cr runtime.Object, service *v1.Service, config *CertConfig) (*v1.Secret, *v1.ConfigMap, *v1.Secret, error) {
+// GenerateCert returns a Secret containing the TLS encryption key and cert,
+// a ConfigMap containing the CA Certificate and a Secret containing the CA key.
```
Comment (AlexNPavel) [inline pkg/tls/tls.go]:
```suggestion
// a ConfigMap containing the CA Certificate, and a Secret containing the CA key.
```

File: pkg/tls/tls.go [hunk 2]
```diff
@@ -139,14 +150,18 @@ type SDKCertGenerator struct {
...
+// GenerateCert returns a error if generation fails.
```
Comment (AlexNPavel) [inline pkg/tls/tls.go]:
```suggestion
// GenerateCert returns an error if generation fails.
```

File: pkg/tls/tls.go [hunk 3]
```diff
@@ -286,99 +303,106 @@ func ToCASecretAndConfigMapName(kind, name string) string {
...
-	}
 	return se, nil
 }
 
-// getCASecretAndConfigMapInCluster gets CA secret and configmap of the given name and namespace.
-// it only returns both if they are found and nil if both are not found. In the case if only one of them is found,
-// then we error out because we expect either both CA secret and configmap exit or not.
+// getCASecretAndConfigMapInCluster gets the CA Secret and ConfigMap of the
+// given name and namespace.
+// getCASecretAndConfigMapInCluster returns a Secret and ConfigMap if found in
```
Comment (AlexNPavel) [inline pkg/tls/tls.go]: These past 3 lines seem a bit redundant. It should be made a bit more concise.

File: pkg/tls/tls.go [hunk 4]
```diff
@@ -286,99 +303,106 @@ func ToCASecretAndConfigMapName(kind, name string) string {
...
+		if !apiErrors.IsNotFound(err) {
+			return nil, nil, err
+		}
 		hasSecret = false
 	}
 
 	if hasConfigMap != hasSecret {
-		// TODO: this case can happen if creating CA configmap succeeds and creating CA secret failed. We need to handle this case properly.
-		return nil, nil, fmt.Errorf("expect either both ca configmap and secret both exist or not exist, but got hasCAConfigmap==%v and hasCASecret==%v", hasConfigMap, hasSecret)
+		// TODO: this case can happen if creating CA ConfigMap succeeds and creating CA Secret failed. We need to handle this case properly.
```
Comment (AlexNPavel) [inline pkg/tls/tls.go]:
```suggestion
		// TODO: this case can happen if creating the CA ConfigMap succeeds and creating the CA Secret failed. We need to handle this case properly.
```

File: pkg/tls/tls.go [hunk 5]
```diff
@@ -139,14 +150,18 @@ type SDKCertGenerator struct {
...
+// a ConfigMap containing the CA Certificate, and a Secret containing the CA key.
+// GenerateCert returns an error if generation fails.
+func (scg *SDKCertGenerator) GenerateCert(cr runtime.Object, service *corev1.Service, config *CertConfig) (*corev1.Secret, *corev1.ConfigMap, *corev1.Secret, error) {
 	if err := verifyConfig(config); err != nil {
 		return nil, nil, nil, err
 	}
 
+	var ownerRefs []metav1.OwnerReference
+	if service != nil {
+		ownerRefs = service.GetOwnerReferences()
```
Comment (estroz) [inline pkg/tls/tls.go]: Can we assume that `service.OwnerReferences` can be applied to TLS objects?

Comment (LiliC) [reply pkg/tls/tls.go]: You mean if the `Service` should be the Owner of the TLS objects?

Comment (estroz) [reply pkg/tls/tls.go]: Should the owners of `service` be the owners of TLS objects.

Comment (LiliC) [reply pkg/tls/tls.go]: Depends how long you want the lifecycle of the TLS objects to be? If I understand correctly the `Secret` that is passed in should be the owner, as generating the TLS objects depends on the Secret object that is passed to the `GenerateCert ` function. So I would say `Service` should be the owner, wdyt?

**Pull request closed without merging**